dts/bindings/counter/nxp,imx-tmr.yaml

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2022 KT-Elektronik, Klaucke und Partner GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP MCUX Quad Timer Channel. Each channel of each quad timer can operate independently
+  and hence will be realized as a separate counter device
+
+compatible: "nxp,imx-tmr"
+
+include: base.yaml
+
+properties:
+  label:
+    required: true
+
+  channel:
+    type: int
+    required: true
+    enum:
+      - 0
+      - 1
+      - 2
+      - 3
+
+  mode:
+    type: int
+    required: true
+    description: counting mode of the timer, see qtmr_counting_mode_t enumerator type
+      of the MCUXpresso SDK
+    enum:
+      - 0
```

Comment:
Can you use a string type for this property? `DT_ENUM_IDX()` can then be used to get the underlying enum value.

Reply:
Thanks, I didn't know that macro yet. It's very useful. Changed accordingly.